pkg/driver/node_test.go

```diff
@@ -1082,6 +1082,92 @@ func TestNodePublishVolume(t *testing.T) {
 		t.Run(tc.name, tc.testFunc)
 	}
 }
+func TestNodeExpandVolume(t *testing.T) {
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+
+	mockMetadata := mocks.NewMockMetadataService(mockCtl)
+	mockMounter := mocks.NewMockMounter(mockCtl)
+
+	awsDriver := &nodeService{
+		metadata: mockMetadata,
+		mounter:  mockMounter,
+		inFlight: internal.NewInFlight(),
+	}
+
+	tests := []struct {
+		name               string
+		request            csi.NodeExpandVolumeRequest
+		expectResponseCode codes.Code
+	}{
+		{
+			name:               "fail missing volumeId",
+			request:            csi.NodeExpandVolumeRequest{},
+			expectResponseCode: codes.InvalidArgument,
+		},
+		{
+			name: "fail missing volumePath",
+			request: csi.NodeExpandVolumeRequest{
+				StagingTargetPath: "/testDevice/Path",
+				VolumeId:          "test-volume-id",
+			},
+			expectResponseCode: codes.InvalidArgument,
+		},
+		{
+			name: "fail volume path not exist",
+			request: csi.NodeExpandVolumeRequest{
+				VolumePath: "./test",
+				VolumeId:   "test-volume-id",
+			},
+			expectResponseCode: codes.Internal,
+		},
+		{
+			name: "Fail validate VolumeCapability",
+			request: csi.NodeExpandVolumeRequest{
+				VolumePath: "./test",
+				VolumeId:   "test-volume-id",
+				VolumeCapability: &csi.VolumeCapability{
+					AccessType: &csi.VolumeCapability_Block{
+						Block: &csi.VolumeCapability_BlockVolume{},
+					},
+					AccessMode: &csi.VolumeCapability_AccessMode{
+						Mode: csi.VolumeCapability_AccessMode_UNKNOWN,
+					},
+				},
+			},
+			expectResponseCode: codes.InvalidArgument,
+		},
+		{
+			name: "Success [VolumeCapability is block]",
+			request: csi.NodeExpandVolumeRequest{
+				VolumePath: "./test",
+				VolumeId:   "test-volume-id",
+				VolumeCapability: &csi.VolumeCapability{
+					AccessType: &csi.VolumeCapability_Block{
+						Block: &csi.VolumeCapability_BlockVolume{},
+					},
+					AccessMode: &csi.VolumeCapability_AccessMode{
+						Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+					},
+				},
+			},
```

Comment:
Can we use `losetup` and `dd` to check the other case of volume_path being a block device?

Reply:
I tried to use losetup and dd to create loop device when setup testing but with no luck. It failed to find an available loop device when I ran it in linux instance.. This should not be a blocker as I tested the functionality e2e on the cluster. Testing code is [here](https://github.com/AndyXiangLi/aws-ebs-csi-driver/blob/skip-block-resizing-test/pkg/driver/node_test.go#L1098), I will follow up on this later.

Reply:
I'm going to approve to unblock you, but yeah let's follow up.